documentation/internal/resource-manager-active-directory.md

```diff
@@ -0,0 +1,292 @@
+- [`Remove-AzADSpCredential`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Remove-AzADSpCredential.md)
+- [`Update-AzADServicePrincipal`](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/help/Update-AzADServicePrincipal.md)
+
+#### User Scenarios
+
+A common scenario is for a user to create a service principal in order to give it specific permissions over a subscription, resource group, or resource in order to automate tasks with the service principal. When the user creates a service principal using the `New-AzADServicePrincipal` cmdlet, there is no way using the Graph SDK to also give the service principal the desired permissions over a given scope; the user would need to go through the portal to manually create the permissions. One of the scenarios we added to this cmdlet was to give users the ability to also create a role assignment for the service principal within this cmdlet using new `-Role` and `-Scope` parameters. If the user provided a value for either parameter, once the service principal was created, a [call to the authorization SDK](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/ActiveDirectory/Cmdlets/NewAzureADServicePrincipalCommand.cs#L346-L380) would be made to assign the specified role to the service principal over the given scope. If neither parameter was provided, or the `-SkipAssignment` switch was set, then a role assignment would not be created. If only `-Role` was provided, `-Scope` would be [set to the user's current subscription](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/ActiveDirectory/Cmdlets/NewAzureADServicePrincipalCommand.cs#L252-L256), and if only `-Scope` was provided, `-Role` would be [set to the "Contributor"](https://github.com/Azure/azure-powershell/blob/master/src/Resources/Resources/ActiveDirectory/Cmdlets/NewAzureADServicePrincipalCommand.cs#L260-L264).
+
+```powershell
+# Create a service principal with no permissions
+New-AzADServicePrincipal @Properties
+
+# Create a service principal with "Reader" permissions over the current subscription
+New-AzADServicePrincipal -Role Reader @Properties
+
+# Create a service principal with "Contributor" permissions over the "MyResourceGroup" resource group
+New-AzADServicePrincipal -Scope /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/MyResourceGroup @Properties
+
+# Create a service principal with "Reader" permissions over the "MyResourceGroup" resource group
+New-AzADServicePrincipal -Role Reader -Scope /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/MyResourceGroup @Properties
+```
+
+_Note_: for parity with Azure CLI, parameters should be added in the future that would allow for the user to create a self-signed certificate and create or retrieve a certificate from a specified Azure KeyVault.
```

Comment:
add in: .Net standard framework does not contain cross-platform libraries for certificate creation, so this would mean identifying platform-specific mechanisms for creating certificates.